tests/validation/tests/v3_api/test_zero_downtime.py

```diff
@@ -0,0 +1,328 @@
+            "Not all Nodes Upgraded Correctly"
+    # Perform Full Restore
+    cluster.restoreFromEtcdBackup(etcdBackupId=backup_info["backup_id"],
+                                  restoreRkeConfig="all")
+    cluster = client.reload(cluster)
+    cluster = validate_cluster_state(
+        client, cluster,
+        check_intermediate_state=True,
+        intermediate_state="updating",
+    )
+    wait_for_node_upgrade(nodes, p_client, cluster, workload, pre_node_ver)
+    # Verify the ingress created before taking the snapshot
+    validate_ingress(p_client, cluster, [backup_info["workload"]], host, path)
+    # Verify the workload created after getting a snapshot does not exist after restore
+    workload_list = p_client.list_workload(uuid=testworkload.uuid).data
+    assert len(workload_list) == 0, "workload shouldn't exist after restore"
+    for node in nodes:
+        assert node["info"]["kubernetes"]["kubeletVersion"] == pre_node_ver, \
+            "Not all Nodes Restored Correctly"
+
+
+def test_zdt_add_worker():
+    client = get_user_client()
+    cluster = namespace["cluster"]
+    p_client = namespace["p_client"]
+    cluster, workload, ingress = validate_cluster_and_ingress(
+        client, cluster,
+        check_intermediate_state=False)
+    # Update Cluster to k8 version + upgrade strategy maxUnavailable worker
+    aws_node = \
+        AmazonWebServices().create_node(random_test_name(HOST_NAME))
+    rke_config = cluster.rancherKubernetesEngineConfig
+    rke_updated_config = rke_config.copy()
+    rke_updated_config["kubernetesVersion"] = K8S_VERSION_UPGRADE
+    rke_updated_config["upgradeStrategy"] = {
+        'drain': True,
+        'maxUnavailableWorker': MAX_UNAVAILABLE_WORKER,
+        'maxUnavailableControlplane': MAX_UNAVAILABLE_CONTROLPLANE,
+        'type': '/v3/schemas/nodeUpgradeStrategy'}
+    cluster = client.update(cluster,
+                            name=cluster.name,
+                            rancherKubernetesEngineConfig=rke_updated_config)
+    nodes = client.list_node(clusterId=cluster.id).data
+    original = len(nodes)
+    # Add worker node
+    docker_run_cmd = \
+        get_custom_host_registration_cmd(client, cluster, ["worker"],
+                                         aws_node)
+    aws_node.roles.append("worker")
+    result = aws_node.execute_command(docker_run_cmd)
+    time.sleep(10)
+    cluster = client.reload(cluster)
+    nodes = client.list_node(clusterId=cluster.id).data
+    # Check Ingress is up during update
+    wait_for_node_upgrade(nodes, p_client, cluster, workload, node_ver)
+    # Validate update has went through
+    assert len(client.list_node(clusterId=cluster.id).data) == original + 1
+    for node in nodes:
+        node = client.reload(node)
+        assert node["info"]["kubernetes"]["kubeletVersion"] == node_ver
+
+
+def validate_node_state(nodes, workload, drain=False, timeout=600,
+                        max_unavailableWorker=1):
```

Comment:
the default is set to in env var 10% but this is where we need the actual count of worker nodes. maybe we can remove the default altogether since it's a helper func

Reply:
there is a default since I combined both the cordon and drain function, only worker nodes/drain needs to a max unavailable to be entered. This is a helper function so max_unavailable must be parameterized as an integer.